src/status_decoder.c

```diff
@@ -0,0 +1,80 @@
+/******************************************************************************
+ * Includes
+ ******************************************************************************/
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+#include <stddef.h>
+#include "status_decoder.h"
+#include "LED_module.h"
+
+/******************************************************************************
+ * Defines
+ ******************************************************************************/
+extern uint16_t crc16(uint8_t *buffer, uint32_t len);
+extern uint8_t ledcontrol_table[LED_MAPPING_END][MAX_STATUS];
+extern uint8_t LEDsetState(int index, int value);
+/******************************************************************************
+ * Typedefs
+ ******************************************************************************/
+
+/******************************************************************************
+ * Global Data
+ ******************************************************************************/
+
+/******************************************************************************
+ * Module Static Data
+ ******************************************************************************/
+
+/******************************************************************************
+ * Local Function Prototypes
+ ******************************************************************************/
+
+/******************************************************************************
+ * Function Definitions
+ ******************************************************************************/
+/**
+ * Decodes a sensor packet received from the On Board Computer
+ *
+ * @param      data  The packet received
+ * @param      buf   The buffer to put the packet data into
+ * @param[in]  len   The length of the buffer
+ *
+ * @return     returns 0 if successful, 1 if otherwise
+ */
+int decodeSensorPacket(DataStatusPacket_t* data, uint8_t* buf, uint32_t len)
```

Comment:
Why is this decoding a sensor packet?  Also, why are we not decoding this byte by byte?